pkg/kubectl/cmd/macro/secret.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package macro
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+// NewCmdCreateSecret groups subcommands to create various types of secrets
+func NewCmdCreateSecret(f *cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "secret",
+		Short: "Create a secret using specified subcommand.",
+		Long:  "Create a secret using specified subcommand.",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Help()
+		},
+	}
+	cmd.AddCommand(NewCmdCreateSecretDockerRegistry(f, cmdOut))
+	cmd.AddCommand(NewCmdCreateSecretOpaque(f, cmdOut))
+	return cmd
+}
+
+const (
+	secretLong = `
+Create a secret based on a file, directory, or specified literal value
+
+Key files can be specified using their file path, in which case a default name will be given to them, or optionally 
+with a name and file path, in which case the given name will be used. Specifying a directory will create a secret 
+using all valid keys in that directory.
+`
+
+	secretExample = `  // Create a new secret named my-secret with keys for each file in folder bar
+  $ kubectl create secret opaque my-secret --from-file=path/to/bar
+
+  // Create a new secret named my-secret with specified keys instead of names on disk
+  $ kubectl create secret opaque my-secret --from-file=ssh-privatekey=~/.ssh/id_rsa --from-file=ssh-publickey=~/.ssh/id_rsa.pub
+
+  // Create a new secret named my-secret with key1=supersecret and key2=topsecret
+  $ kubectl create secret opaque my-secret --from-literal=key1=supersecret --from-literal=key2=topsecret`
```

Comment:
That was what I was implying - kubectl would getenv(VARNAME) to get the
value.  Not sure it is useful, I just get itchy when I see passwords on
commandlines.

On Thu, Nov 19, 2015 at 3:16 PM, Derek Carr <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/macro/secret.go
> <https://github.com/kubernetes/kubernetes/pull/16028#discussion_r45415597>
> :
>
> > +	secretLong = `
> > +Create a secret based on a file, directory, or specified literal value
> > +
> > +Key files can be specified using their file path, in which case a default name will be given to them, or optionally
> > +with a name and file path, in which case the given name will be used. Specifying a directory will create a secret
> > +using all valid keys in that directory.
> > +`
> > +
> > +	secretExample = `  // Create a new secret named my-secret with keys for each file in folder bar
> > +  $ kubectl create secret opaque my-secret --from-file=path/to/bar
> > +
> > +  // Create a new secret named my-secret with specified keys instead of names on disk
> > +  $ kubectl create secret opaque my-secret --from-file=ssh-privatekey=~/.ssh/id_rsa --from-file=ssh-publickey=~/.ssh/id_rsa.pub
> > +
> > +  // Create a new secret named my-secret with key1=supersecret and key2=topsecret
> > +  $ kubectl create secret opaque my-secret --from-literal=key1=supersecret --from-literal=key2=topsecret`
>
> as opposed to --from-literal=key1=$VARNAME?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16028/files#r45415597>.
>